src/main/kotlin/org/wycliffeassociates/otter/jvm/app/ui/takemanagement/view/RecordableFragment.kt

```diff
@@ -0,0 +1,215 @@
+                bottomAnchor = 0.0
+                topAnchor = 0.0
+            }
+        })
+        add(dragContainer)
+    }
+
+    private fun Parent.addDragTakeEventHandlers() {
+        addEventHandler(StartDragEvent.START_DRAG, ::startDrag)
+        addEventHandler(AnimateDragEvent.ANIMATE_DRAG, ::animateDrag)
+        addEventHandler(CompleteDragEvent.COMPLETE_DRAG, ::completeDrag)
+    }
+
+    private fun Parent.addButtonEventHandlers() {
+        addEventHandler(PlayOrPauseEvent.PLAY) {
+            lastPlayOrPauseEvent.set(it)
+        }
+        addEventHandler(DeleteTakeEvent.DELETE_TAKE) {
+            recordableViewModel.deleteTake(it.take)
+        }
+        addEventHandler(EditTakeEvent.EDIT_TAKE) {
+            recordableViewModel.editTake(it)
+        }
+    }
+
+    private fun createSnackBar(pane: Pane) {
+        // TODO: This doesn't actually handle anything correctly. Need to know whether the user
+        // TODO... hasn't selected an editor or recorder and respond appropriately.
+        val snackBar = JFXSnackbar(pane)
+        recordableViewModel.snackBarObservable.subscribe {
+            snackBar.enqueue(
+                JFXSnackbar.SnackbarEvent(
+                    messages["noRecorder"],
+                    messages["addPlugin"].toUpperCase(),
+                    5000,
+                    false,
+                    EventHandler {
+                        audioPluginViewModel.addPlugin(true, false)
+                    })
+            )
+        }
+    }
+
+    private fun createAudioPluginProgressDialog() {
+        // Plugin active cover
+        val dialog = progressdialog {
+            root.addClass(AppStyles.progressDialog)
+            recordableViewModel.contextProperty.toObservable().subscribe { newContext ->
+                graphic = when (newContext) {
+                    TakeContext.RECORD -> AppStyles.recordIcon("60px")
+                    TakeContext.EDIT_TAKES -> AppStyles.editIcon("60px")
+                    null -> null
+                }
+            }
+        }
+        recordableViewModel.showPluginActiveProperty.onChange {
+            Platform.runLater {
+                if (it) dialog.open() else dialog.close()
+            }
+        }
+    }
+
+    private fun Node.bindVisibleToDraggingNodeProperty() {
+        visibleProperty().bind(draggingNodeProperty.booleanBinding { it != null })
+    }
+
+    private fun getPointInRoot(node: Node, pointInNode: Point2D): Point2D {
+        return when (node) {
+            root -> pointInNode
+            else -> getPointInRoot(node.parent, node.localToParent(pointInNode))
+        }
+    }
+
+    private var dragStartDelta = Point2D(0.0, 0.0)
+
+    private fun relocateDragContainer(pointInRoot: Point2D) {
+        val newX = pointInRoot.x - dragStartDelta.x
+        val newY = pointInRoot.y - dragStartDelta.y
+        dragContainer.relocate(newX, newY)
+    }
+
+    private fun startDrag(evt: StartDragEvent) {
```

Comment:
we often aren't going out of our way to abbreviate things; event is already short, can we just use the full word?

Reply:
sure! done